internal/webserver/type/engine.go

```diff
@@ -0,0 +1,99 @@
+package webserver
+
+import (
+	"context"
+	"fmt"
+
+	webscan "github.com/Method-Security/webscan/generated/go"
+	apacheEnumerationModules "github.com/Method-Security/webscan/internal/webserver/type/apache/enumeration"
+	apacheValidationModules "github.com/Method-Security/webscan/internal/webserver/type/apache/validation"
+	nginxEnumerationModules "github.com/Method-Security/webscan/internal/webserver/type/nginx/enumeration"
+	nginxValidationModules "github.com/Method-Security/webscan/internal/webserver/type/nginx/validation"
+)
+
+type ModuleLibrary interface {
+	ModuleRun(target string, config *webscan.WebServerTypeConfig) (*webscan.Attempt, []string)
+	AnalyzeResponse(response *webscan.ResponseUnion) bool
+}
+
+type TypeEngine struct {
+	Library ModuleLibrary
+}
+
+func (be *TypeEngine) Run(ctx context.Context, target string, config *webscan.WebServerTypeConfig) (*webscan.Attempt, []string) {
+	attempt, errs := be.Library.ModuleRun(target, config)
+	return attempt, errs
+}
+
+func TypeLaunch(ctx context.Context, config *webscan.WebServerTypeConfig) (*webscan.WebServerReport, error) {
```

Comment:
This feels like it is maybe something more that should happen on the Engine itself vs a standalone function

Reply:
What do you mean "engine itself"?